c10/metal/indexing.h

```diff
@@ -104,18 +104,23 @@ kernel void unary_strided(
   }
 
 template <typename T, typename F>
-kernel void binary_indexing(
-    constant void* input_ [[buffer(0)]],
-    constant void* other_ [[buffer(1)]],
-    device void* out_ [[buffer(2)]],
-    constant uint3* offsets [[buffer(3)]],
-    uint tid [[thread_position_in_grid]]) {
-  auto out =
-      (device result_of<F, T, T>*)((device uint8_t*)out_ + offsets[tid].x);
-  auto input = (constant T*)((constant uint8_t*)input_ + offsets[tid].y);
-  auto other = (constant T*)((constant uint8_t*)other_ + offsets[tid].z);
+kernel void binary_strided(
+    constant T* input [[buffer(0)]],
+    constant T* other [[buffer(1)]],
+    device result_of<F, T, T>* output [[buffer(2)]],
+    constant long* sizes [[buffer(3)]],
+    constant long* input_strides [[buffer(4)]],
+    constant long* other_strides [[buffer(5)]],
+    constant long* output_strides [[buffer(6)]],
+    constant uint& ndim [[buffer(7)]],
+    uint index [[thread_position_in_grid]]) {
   F f;
-  *out = f(*input, *other);
+  int pos[max_ndim];
```

Comment:
I see max_ndim is defined as 16 at the top of this file.
Where are we checking that this is the maximum allowed number of dimensions of the intputs/output for this path to be taken?

Reply:
Yes, you can't create more than 16 dim tensor with MPS, see 
https://github.com/pytorch/pytorch/blob/0a60a0cad4132c88e7338469aa40b45593c07181/aten/src/ATen/mps/EmptyTensor.cpp#L38